src/components/tests/Outline.js

```diff
@@ -59,4 +76,26 @@ describe("Outline", () => {
     await symbolsPromise;
     expect(component).toMatchSnapshot();
   });
+
+  it("should select a line of code in the current file on click", async () => {
+    const component = shallow(new OutlineComponent({ selectSource }));
+    const sourceId = "id";
+    const selectedSource = {
+      get: () => sourceId
+    };
+    const startLine = 12;
+
+    symbolDeclarations.functions[0] = {
+      id: "my_example_function1:21",
+      value: "my_example_function1",
+      location: generateFuncLocation(startLine)
+    };
+
+    component.setProps({ sourceText, selectedSource });
```

Comment:
could we pass selectedSource in when we initially render the component? 

Perhaps not based on how we're fetching the symbols, which is a shame i suppose. And if that's the case, we probably move symbols to a reducer so that we can have the symbols passed down alongside the source :) (we can do this in a new issue ofcourse)